src/server/services/wms/qgswmsrenderer.h

```diff
@@ -195,6 +195,12 @@ namespace QgsWms
        */
       void configureMapSettings( const QPaintDevice *paintDevice, QgsMapSettings &mapSettings, bool mandatoryCrsParam = true );
 
+      /**
+       * Configures QgsRenderContext according to the WMS parameters and default settings. Used, for example, when no mapSettings are available.
+       * \param context render context keeping info about the scale
+       */
+      void configureDefaultRenderContext( QgsRenderContext &context );
```

Comment:
Could this be changed as follows, I think it would be more intuitive to use?

```suggestion
      QgsRenderContext configureDefaultRenderContext( QPainter *painter );
```

Reply:
Why having the painter here? Would call `QgsRenderContext::fromQPainter` in this function? On one of the use cases the painter would be null.